desk.go

```diff
@@ -153,5 +207,6 @@ func NewDesktop(app fyne.App, wm WindowManager, icons ApplicationProvider) Deskt
 // fyne/test package.
 func NewEmbeddedDesktop(app fyne.App, icons ApplicationProvider) Desktop {
 	instance = &deskLayout{app: app, icons: icons, settings: NewDeskSettings()}
+	instance.(*deskLayout).screens = instance.(*deskLayout)
```

Comment:
Because in the case of an embedded desktop - the deskLayout won't have access to x11wm information

So depending on if it is embedded desktop vs regular - we set which interface to use - which is cleaner in the rest of the code to say deskLayout.screens rather than if (embedded) everywhere.